src/search-injection/components/Results.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+import Dropdown from './Dropdown'
+import ResultItem from './ResultItem'
+import RemovedText from './RemovedText'
+import * as constants from '../constants'
+import { getLocalStorage, setLocalStorage } from '../utils'
+
+import styles from './Results.css'
+
+class Results extends React.Component {
+    static propTypes = {
+        results: PropTypes.arrayOf(PropTypes.object).isRequired,
+        len: PropTypes.number.isRequired,
+        rerender: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super(props)
+        this.seeMoreResults = this.seeMoreResults.bind(this)
+        this.toggleHideResults = this.toggleHideResults.bind(this)
+        this.toggleDropDown = this.toggleDropDown.bind(this)
+        this.removeResults = this.removeResults.bind(this)
+        this.undoRemove = this.undoRemove.bind(this)
+        this.changePosition = this.changePosition.bind(this)
+    }
+
+    state = {
+        hideResults: true,
+        dropdown: false,
+        removed: false,
+        position: null,
+    }
+
+    async componentDidMount() {
+        const hideResults = await getLocalStorage(
+            constants.HIDE_RESULTS_KEY,
+            false,
+        )
+        const position = await getLocalStorage(constants.POSITION_KEY, 'above')
+        this.setState({
+            hideResults,
+            position,
+        })
+    }
+
+    renderResultItems() {
+        const resultItems = this.props.results.map((result, i) => (
+            <ResultItem key={i} {...result} />
+        ))
+        return resultItems
+    }
+
+    seeMoreResults() {
+        // Create a new tab with the query overview URL
+        const query = new URL(location.href).searchParams.get('q')
+        const url = `${constants.OVERVIEW_URL}?query=${query}`
+        const message = {
+            action: 'openOverviewURL',
+            url,
+        }
+        browser.runtime.sendMessage(message)
+    }
+
+    async toggleHideResults() {
+        // Toggles hideResults (minimize) state
+        // And also, sets dropdown to false
+        const toggled = !this.state.hideResults
+        await setLocalStorage(constants.HIDE_RESULTS_KEY, toggled)
+        this.setState({
+            hideResults: toggled,
+            dropdown: false,
+        })
+    }
+
+    toggleDropDown() {
+        this.setState(state => ({
```

Comment:
You're using both reducer and object versions of `this.setState`; it would be good to be consistent with which you use. In the rest of the code base we're trying to use the reducer (function) taking version. Really important to note though, if you use the reducer version like here: the returned value of the callback will be the new state.

```js
this.setState(state => ({ dropdown: !state.dropdown }))
```
The above will unset all the other state keys, like `hideResults`, meaing they will be `undefined` whenever they get accessed via `this.state`. That may be your intention here though, but it's often a gotcha with `this.setState`. 

The following would ensure `dropdown` state is updated, and all other state keys are left alone:

```js
this.setState(state => ({ ...state, dropdown: !state.dropdown }))
```